pandas/tools/merge.py

```diff
@@ -1264,13 +1264,21 @@ def flip(xs):
 
         # a "by" parameter requires special handling
         if self.left_by is not None:
```

Comment:
I guess I could leave a comment since it also affects the new line 1276. Basically, the left and right "by" parameters have to be the same length. (I should also add verification of this starting at line 1157.) Because they are the same length, checking for one means I can assume the other. That's why I only bother with a left conditional.